### DISCUSSION TITLE
CONEXÃO COM Wi-Fi

### DISCUSSION THREAD

**Discussion by caique-dourado** [Flutterando/forum#70] [Discussões gerais]
Estou criando uma aplicação para se conectar a uma rede de wi-fi, usei alguns plugins, mas nenhum da certo.
conforme o código abaixo e as permissões.
Mas quando me conecto, não da sinal de internet, o que fazer?.
PERMISSÕES:
  "android.permission.ACCESS_WIFI_STATE"
  "android.permission.CHANGE_WIFI_STATE"
  "android.permission.CHANGE_NETWORK_STATE"
  "android.permission.ACCESS_FINE_LOCATION"

CODIGO:
`import 'package:flutter/material.dart';
import 'package:plugin_wifi_connect/plugin_wifi_connect.dart';

void main() {
  runApp(const MyWidget());
}

class MyWidget extends StatefulWidget {
  const MyWidget({super.key});

  @override
  State<MyWidget> createState() => _MyWidgetState();
}

class _MyWidgetState extends State<MyWidget> {
  connect() {
    PluginWifiConnect.connectToSecureNetwork(
      'nomeSSID',
      'SENHADOWIFI',
      saveNetwork: true,
    );
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Center(
            child: SizedBox(
          child: TextButton(
              onPressed: () {
                setState(() {
                  connect();
                });
              },
              child: const Text('CONECTAR')),
        )),
      ),
    );
  }
}
`

**Comment by eduardoflorence** [comment]
Você poderia utilizar este package - https://pub.dev/packages/permission_handler - para ter certeza que as permissões foram dadas pelo usuário. Exemplo:

```dart
import 'package:permission_handler/permission_handler.dart';
...
if (await Permission.location.request().isGranted) {
    // connect
}
```

**Reply by caique-dourado** [reply]
A permissões estão todas okay. Estava lendo a doc e o SDK >29+ em geral não está permitindo conexão direta com outros dispositivos!
